src/pjpersist/serialize.py

```diff
@@ -356,15 +356,17 @@ def get_state(self, obj, pobj=None):
             # Same as for sequences, make sure that the contained values are
             # properly serialized.
             # Note: A big constraint in JSONB is that keys must be strings!
-            has_non_string_key = False
+            has_non_compliant_key = False
             data = []
             for key, value in obj.items():
                 data.append((key, self.get_state(value, pobj)))
-                has_non_string_key |= not isinstance(key, six.string_types)
-                if (not isinstance(key, six.string_types) or '\0' in key):
-                    has_non_string_key = True
-            if not has_non_string_key:
-                # The easy case: all keys are strings:
+                if (not isinstance(key, six.string_types) or  # non-string
+                    '\0' in key or  # string having a 0x00
+                    key=='dict_data'  # a key with our special marker
```

Comment:
Huh? what is `\0`? what is `dict_data`? What if there is `0x1` character in the string? What if there is `0x2` character? What if  there is "list_data" in it?

Reply:
`\0` is a chr(0) looks like it's mongo legacy, with PG/JSON as investigation shows you just cannot store a chr(0) be it in a key or just a simple string

Checking for `dict_data` is there to avoid failing in `get_object` as this is a special marker when encoding non-string keys.
See `doctest_ObjectWriter_get_state_mappings` "Another edge case is when the input has a `dict_data` key"

Reply:
We can't really change this `dict_data` encoding, because then we would need a migration or BBB code.

Reply:
I don't understand why special marker is called `dict_data`. Shouldn't it be called `special_marker` instead?

Reply:
Yes that's why it's there, but do we want migrations? Look at the recent push, I made it a global.

Reply:
Why are you talking about migrations?

Reply:
Because that weirdo `dict_data` gets persisted to the database.
See `doctest_ObjectWriter_get_state_mappings` how state looks.

Reply:
Well, `list_data` will also persist itself if it is part of the object state. Should we also handle it here?